pkg/cli/discover.go

```diff
@@ -99,3 +135,68 @@ func checkExtEndPoint(config searchConfig) bool {
 
 	return containsGQLQuery(queryResponse.Data.Schema.QueryType.Fields, "ImageList")
 }
+
+func CheckExtEndPointQuery(config searchConfig, requiredQueries ...GQLQuery) error {
+	username, password := getUsernameAndPassword(*config.user)
+	ctx := context.Background()
+
+	discoverEndPoint, err := combineServerAndEndpointURL(*config.servURL, fmt.Sprintf("%s%s",
+		constants.RoutePrefix, constants.ExtOciDiscoverPrefix))
+	if err != nil {
+		return err
+	}
+
+	discoverResponse := &distext.ExtensionList{}
+
+	_, err = makeGETRequest(ctx, discoverEndPoint, username, password, *config.verifyTLS,
+		*config.debug, &discoverResponse, config.resultWriter)
+	if err != nil {
+		return err
+	}
+
+	searchEnabled := false
+
+	for _, extension := range discoverResponse.Extensions {
+		if extension.Name == "_zot" {
+			for _, endpoint := range extension.Endpoints {
+				if endpoint == constants.FullSearchPrefix {
+					searchEnabled = true
+				}
+			}
+		}
+	}
+
+	if !searchEnabled {
+		return fmt.Errorf("%w: search extension gql endpoints", zerr.ErrExtensionNotEnabled)
+	}
+
+	searchEndPoint, _ := combineServerAndEndpointURL(*config.servURL, constants.FullSearchPrefix)
+
+	schemaQuery := `
+        {
+            __schema() {
+                queryType {
+                    fields {
+                        name
+						args {
```

Comment:
is this indentation intentional?

Reply:
I believe the comment was about mixing tabs with spaces in this specific case

Reply:
@laurentiuNiculae can you please double check this?